tiledb/sm/tile/tile_metadata_generator.cc

```diff
@@ -173,9 +173,9 @@ ByteVec Sum<T, uint64_t>::sum_nullable(Tile* tile, Tile* tile_validity) {
   ByteVec ret(8, 0);
 
   // Get pointers to the data and cell num.
-  auto values = static_cast<T*>(tile->buffer()->data());
-  auto validity_values = static_cast<uint8_t*>(tile_validity->buffer()->data());
-  auto cell_num = tile->buffer()->size() / sizeof(T);
+  auto values = static_cast<T*>(tile->data());
```

Comment:
I picked this statement at random to make this comment. We would benefit from a function `template<class T> T* data_as()` to package up the cast to make the code easier to read.